pkg/resources/webhook.go

```diff
@@ -0,0 +1,281 @@
+	}
+
+	return diags
+}
+
+func resourceWebhookCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	c := m.(*dbt_cloud.Client)
+
+	// Warning or errors can be collected in a slice type
+	var diags diag.Diagnostics
+
+	webhookId := ""
+	name := d.Get("name").(string)
+	description := d.Get("description").(string)
+	clientUrl := d.Get("client_url").(string)
+	eventTypes := d.Get("event_types").([]interface{})
+	jobIds := d.Get("job_ids").([]interface{})
+	active := d.Get("active").(bool)
+
+	typedEventTypes := []string{}
+	for _, eventType := range eventTypes {
+		typedEventTypes = append(typedEventTypes, eventType.(string))
+
+	}
+	typedJobIds := []int{}
+	for _, jobId := range jobIds {
+		typedJobIds = append(typedJobIds, jobId.(int))
+	}
+
+	w, err := c.CreateWebhook(
+		webhookId,
+		name,
+		description,
+		clientUrl,
+		typedEventTypes,
+		typedJobIds,
+		active,
+	)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.SetId(w.WebhookId)
+
+	// we have to set it at the create level
+	if err := d.Set("hmac_secret", w.HmacSecret); err != nil {
+		return diag.FromErr(err)
+	}
+
+	resourceWebhookRead(ctx, d, m)
+
+	return diags
+}
+
+func resourceWebhookUpdate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	c := m.(*dbt_cloud.Client)
+	webhookId := d.Id()
+
+	if d.HasChange("name") ||
+		d.HasChange("description") ||
+		d.HasChange("client_url") ||
+		d.HasChange("event_types") ||
+		d.HasChange("job_ids") ||
+		d.HasChange("active") {
+		webhookRead, err := c.GetWebhook(webhookId)
+
+		jobIdsWrite := make([]int, len(webhookRead.JobIds))
+		for i, step := range webhookRead.JobIds {
+			jobIdsWrite[i], _ = strconv.Atoi(step)
+		}
+
+		webhook := dbt_cloud.WebhookWrite{
+			WebhookId:   webhookId,
+			Name:        webhookRead.Name,
+			Description: webhookRead.Description,
+			ClientUrl:   webhookRead.ClientUrl,
+			EventTypes:  webhookRead.EventTypes,
+			JobIds:      jobIdsWrite,
+			Active:      webhookRead.Active,
+		}
+
+		if err != nil {
+			return diag.FromErr(err)
+		}
+
+		if d.HasChange("name") {
+			name := d.Get("name").(string)
+			webhook.Name = name
+		}
+		if d.HasChange("description") {
+			description := d.Get("description").(string)
+			webhook.Description = description
+		}
+		if d.HasChange("client_url") {
+			clientUrl := d.Get("client_url").(string)
+			webhook.ClientUrl = clientUrl
+		}
+		if d.HasChange("event_types") {
+			eventTypes := make([]string, len(d.Get("event_types").([]interface{})))
+			for i, step := range d.Get("event_types").([]interface{}) {
+				eventTypes[i] = step.(string)
+			}
+			webhook.EventTypes = eventTypes
+		}
+		if d.HasChange("job_ids") {
+			jobIDs := make([]int, len(d.Get("job_ids").([]interface{})))
+			for i, step := range d.Get("job_ids").([]interface{}) {
+				jobIDs[i] = step.(int)
+			}
+			webhook.JobIds = jobIDs
+		}
+		if d.HasChange("active") {
+			active := d.Get("active").(bool)
+			webhook.Active = active
```

Comment:
just to check, this doesn't delete it if it's set to not active like some of the other endpoints, right?

Reply:
The API doesn't actually allow setting active to `false` and we need to `DELETE` it instead. I remove this from triggering a change to the resource.